demos/mqtt/mqtt_demo_lightweight/demo_config.h

```diff
@@ -53,7 +53,9 @@
  * This demo uses the Mosquitto test server. This is a public MQTT server; do not
  * publish anything sensitive to this server.
  */
-#define BROKER_ENDPOINT           "test.mosquitto.org"
+#ifndef BROKER_ENDPOINT
+    #define BROKER_ENDPOINT    "test.mosquitto.org"
```

Comment:
The cautionary language on lines 53-55 is not sufficient here.  From the [test.mosquitto.org documentation](http://test.mosquitto.org/), the test.mosquitto.org broker _"...will often be running unreleased or experimental code and may not be as stable as you might hope."_. Additional cautionary language on their website asks _"...please do not abuse or rely upon it for anything of importance."_.

I would recommend removing references to test.mosquitto.org from this (and any other) MQTT demos.  If someone wishes to use it, they of course may do so.  However the best way to test this by running a broker locally, and we can easily provide instructions for setting up mosquitto or another broker to run locally.

Reply:
Such directions are available in the readme too.

Reply:
+1 to Gary's comment.

Reply:
It's a good suggestion to not _suggest_ `test.mosquitto.org` in our demos. I had added instructions on the REAMDE for a way to setup the Mosquitto broker locally: https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/development#installing-mosquitto-to-run-mqtt-demos-locally

We could update the comments on our `demo_config.h` to point to these instructions